### PR TITLE
- Prefer the VM ID, which we can get from the KVP pool information

### PR DESCRIPTION
**Pull request by rjschwei**
populated by the kvp daemon and is extracted from the host over a
  randomly generated ID as the subscription ID.

### CODE REVIEW

File: azuremetadata [hunk 1]
```diff
@@ -44,36 +46,51 @@ try:
...
         # generate subscriptionId if it's unavailable
         if 'subscriptionId' not in util.available_params:
-            if len(util.available_params) > 1:
-                # if API response succeeded, but no subscriptionId is defined -- generate random ID
-                data['subscriptionId'] = 'classic-{}'.format(random.randint(0, 1e9))
+            # try the kvp data or generate random ID
+            if os.path.exists('/var/lib/hyperv/.kvp_pool_3'):
+                kvp_key_size = 512
+                kvp_value_size = 2048
+                kvp_pool_3 = open('/var/lib/hyperv/.kvp_pool_3', 'rb')
```
Comment (jesusbv) [inline azuremetadata]: It would be better to add `with open...as `
Also, the file path could go in a variable as it is used a couple of times.

**Comment by rjschwei** [comment]
The current implementation is insufficient to support AHB w.r.t. VMs running in ASM (Classic). Still need an identifier from Azure that we can insert accordingly.

**Comment by rjschwei** [comment]
Ready now. just in case @ikapelyukhin gets bored on vacation and wants to do some code review ;)

File: lib/azuremetadata/azuremetadata.py [hunk 2]
```diff
@@ -80,6 +80,12 @@ def list_api_versions(self):
         # considered all available APIs
         return self._get_api_newest_versions()
 
+    def set_api_version(self, api_version):
```
Comment (jesusbv) [inline lib/azuremetadata/azuremetadata.py:83]: This method should be used in `__init__` as well, that change should be a different PR IMO.

**Review by jesusbv — APPROVED** [review]
LGTM

**Review by ikapelyukhin — COMMENTED** [review]
The implementation is getting weirder with each day :slightly_smiling_face:
